wrapper/src/main/java/software/amazon/jdbc/ConnectionProviderManager.java

```diff
@@ -0,0 +1,72 @@
+package software.amazon.jdbc;/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.util.Properties;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class ConnectionProviderManager {
+
+  private static ConnectionProvider connProvider = null;
+  private ConnectionProvider defaultProvider;
+  private static final ReentrantLock lock = new ReentrantLock();
```

Comment:
In this particular case I think ReentrantReadWriteLock could be better. We shouldn't care about it now. It's a POC.